common/flatpak-run.c

```diff
@@ -3854,7 +3942,7 @@ flatpak_run_app (FlatpakDecomposed *app_ref,
 
   if (!flatpak_run_add_app_info_args (bwrap,
                                       app_files, app_deploy_data, app_extensions,
-                                      runtime_files, runtime_deploy_data, runtime_extensions,
```

Comment:
I think this is right, but we should perhaps add a new field to the flatpak-info file so you can tell what /usr is.

Reply:
Yes, I think that makes sense. I think the Steam-launched subsandbox should look like this:

* "official" runtime: `org.freedesktop.Platform/x86_64/20.08` (or whatever version Steam is using by the time we land this)
* replacement `/usr`: `/home/me/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier/var/tmp-abcdef/usr`

instead of pretending the runtime is `com.valvesoftware.SteamRuntime.Platform/x86_64/soldier` - which it sort of is, but Flatpak has no good way to know it can trust a string supplied by pressure-vessel.

Reply:
In the next revision of this branch I'm going to make it look something like this:

```
[Application]
name=com.valvesoftware.Steam
runtime=runtime/org.freedesktop.Platform/x86_64/20.08

[Instance]
app-path=
original-app-path=/var/lib/flatpak/app/.../files
runtime-path=/home/me/.../SteamLinuxRuntime_soldier/var/tmp-abcdef/usr
original-runtime-path=/var/lib/flatpak/runtime/.../files
```

where `app-path=` denotes "an empty directory on a tmpfs".

Similarly, `flatpak run --app-path=/tmp/app` would come out as `app-path=/tmp/app`.

The other possibility would be to set `app-path` and `runtime-path` to the "official" app and runtime files, and have a new `replacement-app-path` and `replacement-usr-path`, like this:

```
[Application]
name=com.valvesoftware.Steam
runtime=runtime/org.freedesktop.Platform/x86_64/20.08

[Instance]
app-path=/var/lib/flatpak/app/.../files
replacement-app-path=
runtime-path=/var/lib/flatpak/runtime/.../files
replacement-usr-path=/home/me/.../SteamLinuxRuntime_soldier/var/tmp-abcdef/usr
```

but I think that has a worse compatibility story for existing versions of `xdg-desktop-portal`, which is the only user I can find of those fields.